### PR TITLE
fix: 🐛 Docker if image exists check only on non-CI

### PR DESCRIPTION
**Pull request by Dunky13**

### CODE REVIEW

File: binzx/otomi [hunk 1]
```diff
@@ -300,8 +302,11 @@ fi
 
 mkdir -p /tmp/otomi
 container_name='otomi-core-dev-container'
-count=$(docker ps -a | grep $container_name | wc -l) >/dev/null
-count=$(($count))
+count=0
+if [ -n "$has_docker" ]; then
+  count=$(docker ps -a | grep $container_name | wc -l) >/dev/null
+  count=$(($count))
```
Comment (Morriz) [inline binzx/otomi:308]:
```suggestion
  count=$((docker ps -a | grep $container_name | wc -l)) >/dev/null
```

**Review by Morriz — CHANGES_REQUESTED** [review]

File: binzx/otomi (hunk 1 above)
Comment (Dunky13) [reply binzx/otomi:308]: Nope, doesn't work:
```
➜ count=$((docker ps -a | grep $container_name | wc -l)) >/dev/null
zsh: bad math expression: operator expected at `ps -a | gr...'
```

File: binzx/otomi [hunk 2]
```diff
@@ -43,6 +43,7 @@ silent() {
 }
 
 docker_tag_exists() {
+  [ -n "$CI" ] && return 0
```
Comment (Morriz) [inline binzx/otomi:46]: why is this added?

**Review by Morriz — CHANGES_REQUESTED** [review]

File: binzx/otomi (hunk 2 above)
Comment (Dunky13) [reply binzx/otomi:46]: To skip when running in CI mode

Comment (Morriz) [reply binzx/otomi:46]: is that needed? and here in this function? out of place and confusing

Comment (Morriz) [reply binzx/otomi:46]: please create conditions where they apply, so it stays maintainable

File: binzx/otomi [hunk 3]
```diff
@@ -300,8 +302,10 @@ fi
 
 mkdir -p /tmp/otomi
 container_name='otomi-core-dev-container'
-count=$(docker ps -a | grep $container_name | wc -l) >/dev/null
-count=$(($count))
+count=0
+if [ -n "$has_docker" ]; then
+  count=$(($(docker ps -a | grep -c $container_name))) >/dev/null
+fi
```
Comment (Morriz) [inline binzx/otomi:308]:
```suggestion
[ -n "$has_docker" ] &&  count=$(($(docker ps -a | grep -c $container_name))) >/dev/null
```
